packages/block-editor/src/components/block-patterns-list/index.js

```diff
@@ -206,43 +226,38 @@ function BlockPatternsList(
 	},
 	ref
 ) {
-	const compositeStore = useCompositeStore( { orientation } );
-	const { setActiveId } = compositeStore;
-
-	useEffect( () => {
-		// We reset the active composite item whenever the
-		// available patterns change, to make sure that
-		// focus is put back to the start.
-		setActiveId( undefined );
-	}, [ setActiveId, shownPatterns, blockPatterns ] );
```

Comment:
Moved this effect to a child of `Composite` (called `BlockPatternsListContentWrapper`), which can access the `store` via content.